youtube_dl/extractor/francetv.py

```diff
@@ -128,17 +128,37 @@ def sign(manifest_url, manifest_id):
 
         is_live = None
 
-        formats = []
+        videos = []
+
         for video in info['videos']:
             if video['statut'] != 'ONLINE':
                 continue
+            if not video['url']:
+                continue
+            videos.append(video)
+
+        if not videos:
+            for device_type in ['desktop', 'mobile']:
+                fallback_info = self._download_json(
+                    'https://player.webservices.francetelevisions.fr/v1/videos/%s' % video_id,
+                    video_id, 'Downloading fallback %s video JSON' % device_type, query={
+                        'device_type': device_type,
+                        'browser': 'chrome',
+                    }, fatal=False)
+
+                if 'video' in fallback_info:
```

Comment:
still would break the extraction if one request fails(with `fatal=False` the return value of `_download_json` would be `None` and you can the `in` operator with `NoneType`).
```suggestion
                if fallback_info and fallback_info.get('video'):
```